arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -448,6 +470,19 @@ public String toString() {
 
     @Override
     public void setBatchSize(int batchSize) {
+      this.batchSize = (batchSize == 0) ? DEFAULT_BATCH_SIZE : batchSize;
+      this.nulls = newNullabilityHolder(batchSize);
```

Comment:
Shouldn't we only do this if the batch size changed and not every time this is called?